lib/events/index.js

```diff
@@ -958,14 +959,32 @@ export default function (self) {
       y = self.style.minRowHeight;
     }
     if (
-      self.dispatchEvent('resizecolumn', {
+      self.dispatchEvent(resizingColumn ? 'resizecolumn' : 'resizerow', {
         x: x,
         y: y,
+        width: x,
+        height: y,
+        index: resizingColumn
```

Comment:
@velitasali, apologies, late to the party here. The key-name `index` here is not particular descriptive. I think using `columnIndex` and `rowIndex` for `resizecolumn` and `resizerow` events respectively makes it more explicit what you're dealing with as a consumer of the api/events, e.g. `{ ... index: 3 }` vs `{ ... rowIndex: 3 }`. 

What do you think?

Reply:
I'm inclined to agree! I wonder if we should then always have both `rowIndex` _and_ `columnIndex` present (where for instance `columnIndex: null` if the event is `resizerow`) or only one of the two keys, depending on the type of event.

My instinct says _both_, but I can't quite articulate why.

Reply:
@mdebrauw I am not sure. If it is going to help users to have an easier time, we can go that way (explicitly specifying `columnIndex` or `rowIndex`). Other than that, the current approach should require less boilerplate, imo.

Reply:
If boilerplate when _writing_ internal canvas-datagrid code is a concern, we could consider a helper function? I very much liked @hangxingliu's approach with [getContextOfSelectionEvent](https://github.com/TonyGermaneri/canvas-datagrid/pull/498/files#diff-4430e2d62644251165d972c62b389ef52b749cbd19c3ee418888833190317801R382), as a way of getting/providing the data needed to emit an event.

```js
self.dispatchEvent('selectionchanged', self.getContextOfSelectionEvent());
```

This should reduce the boilerplate needed, I think. Even so, I tend towards preferring the code more explicit, clearer to debug and easier read for future contributors — but perhaps at the risk of being overly verbose/boilerplate-y. But it's a balancing act, for sure.

Reply:
I see, but I am not sure if there is a need for that here. I mean, is there a function that we can offload the parameters provided by the event like `getContextOfSelectionEvent` does?

Reply:
Ah right, I see what you mean. Sorry, never mind then :-)

But I don't think I'm understanding the "less boilerplate" argument — we want to be passing the row or column index for each `resizerow` or `resizecolumn` event, correct? So then it makes no difference, in terms of boilerplate, whether it's called `index` instead of `rowIndex` or `columnIndex` ?

Reply:
Now that I think about it, I am also for going with `rowIndex` and `columnIndex` since users can assign the same function to  both `resizerow` and `resizecolumn`. For the index that is not available, we can give -1 or undefined, which (I think) is more predictable.

Reply:
Ah yes, I didn't even consider using the same function — good one. `undefined` sounds great!